Gems/Terrain/Code/Tests/TerrainPhysicsColliderTests.cpp

```diff
@@ -271,8 +298,34 @@ TEST_F(TerrainPhysicsColliderComponentTest, TerrainPhysicsColliderReturnsRelativ
     AZ::Vector2 mockHeightResolution = AZ::Vector2(1.0f);
 
     NiceMock<UnitTest::MockTerrainDataRequests> terrainListener;
-    ON_CALL(terrainListener, GetHeightFromFloats).WillByDefault(Return(mockHeight));
     ON_CALL(terrainListener, GetTerrainHeightQueryResolution).WillByDefault(Return(mockHeightResolution));
+    ON_CALL(terrainListener, ProcessHeightsFromRegion).WillByDefault(
+        [mockHeight](const AZ::Aabb& inRegion, const AZ::Vector2& stepSize,
+            AzFramework::Terrain::SurfacePointRegionFillCallback perPositionCallback,
+            AzFramework::Terrain::TerrainDataRequests::Sampler sampleFilter)
+        {
+            if (!perPositionCallback)
+            {
+                return;
+            }
+
+            const size_t numSamplesX = aznumeric_cast<size_t>(ceil(inRegion.GetExtents().GetX() / stepSize.GetX()));
+            const size_t numSamplesY = aznumeric_cast<size_t>(ceil(inRegion.GetExtents().GetY() / stepSize.GetY()));
+
+            AzFramework::SurfaceData::SurfacePoint surfacePoint;
+            for (size_t y = 0; y < numSamplesY; y++)
+            {
+                float fy = aznumeric_cast<float>(inRegion.GetMin().GetY() + (y * stepSize.GetY()));
+                for (size_t x = 0; x < numSamplesX; x++)
+                {
+                    bool terrainExists = false;
+                    float fx = aznumeric_cast<float>(inRegion.GetMin().GetX() + (x * stepSize.GetX()));
+                    surfacePoint.m_position.Set(fx, fy, mockHeight);
+                    perPositionCallback(x, y, surfacePoint, terrainExists);
+                }
+            }
+        }
```

Comment:
Looks equivalent to above, should be able to put it in a separate function.

Reply:
In each case, the lambda capture is different.

Reply:
Sure, but the lambda can just call another function that does this work. Only internal difference is the z position which could be passed in as mockHeight or 0.

Reply:
For this case and the one above, yes. The case below this one has surface tag captures as well. I suppose I could make it a surfaceTag list instead and pass that into the function.